pkg/agent/worker.go

```diff
@@ -254,41 +270,52 @@ func (w *Worker) Close() {
 		return
 	}
 
-	w.Logger.Infow("closing worker")
+	w.logger.Infow("closing worker")
 
 	close(w.closed)
 	w.cancel()
 	_ = w.conn.Close()
 	w.mu.Unlock()
+
+	if w.registered.Load() {
+		w.handler.HandleWorkerDeregister(w)
+	}
 }
 
 func (w *Worker) HandleMessage(req *livekit.WorkerMessage) {
 	switch m := req.Message.(type) {
 	case *livekit.WorkerMessage_Register:
-		go w.handleRegister(m.Register)
+		w.handleRegister(m.Register)
```

Comment:
Can you comment on the reason to make these synchronous? (Or on why these were initially made to run in a separate goroutine?)

Reply:
i can't speculate on why this used goroutines but
* making it synchronous eliminates a whole class of difficult to reproduce bugs
* a single thread should have no problem keeping up with a worker and there are likely to be many more worker connections than cpu cores
* most of the operations are too small to justify the overhead of starting a goroutine
* none of the handlers block on io except during server registration/deregistration which only happens once per namespace/job type
* all of the operations require holding the worker mutex to perform their tasks so there's very little opportunity for parallelizing work